app/com/linkedin/drelephant/tez/fetchers/TezFetcher.java

```diff
@@ -255,31 +261,34 @@ private long getDagEndTime(URL url) throws IOException, AuthenticationException
     }
 
     private void getTaskDataAll(URL vertexListUrl, String dagId, List<TezTaskData> mapperList,
-                  List<TezTaskData> reducerList) throws IOException, AuthenticationException {
+                                List<TezTaskData> reducerList, List<TezTaskData> scopeTaskList) throws IOException, AuthenticationException {
 
       JsonNode rootVertexNode = ThreadContextMR2.readJsonNode(vertexListUrl);
       JsonNode vertices = rootVertexNode.path("entities");
       boolean isMapVertex = false;
-
+      String vertexType = null;
       for (JsonNode vertex : vertices) {
         String vertexId = vertex.get("entity").getTextValue();
         String vertexClass = vertex.path("otherinfo").path("processorClassName").getTextValue();
-
-        if (vertexClass.equals("org.apache.hadoop.hive.ql.exec.tez.MapTezProcessor"))
+        URL tasksByVertexURL = getTaskListByVertexURL(dagId, vertexId);
+        if (vertexClass.equals("org.apache.hadoop.hive.ql.exec.tez.MapTezProcessor")) {
           isMapVertex = true;
-        else if (vertexClass.equals("org.apache.hadoop.hive.ql.exec.tez.ReduceTezProcessor"))
+          getTaskDataByVertexId(tasksByVertexURL, dagId, vertexId, mapperList,isMapVertex);
```

Comment:
I think you can get rid of the variable, `isMapVertex`, and directly use `true` here.